src/fabric/src/fabric_doc_update.erl

```diff
@@ -318,13 +337,90 @@ group_docs_by_shard(DbName, Docs) ->
         )
     ).
 
-append_update_replies([], [], DocReplyDict) ->
+%% use 'lowest' node that hosts this shard range as leader
+is_leader(Worker, Workers) ->
+    Worker == lists:min([W || W <- Workers, W#shard.range == Worker#shard.range]).
```

Comment:
Wonder what's being used to decide the "lowest" here. Workers have refs in them, wouldn't those induce a random order, that wouldn't work for the general idea of all coordinators picking one "leader" then independently.

If we want to decide based on node name or node name + shard range hash it might be better to make that explicit.

Reply:
agreed